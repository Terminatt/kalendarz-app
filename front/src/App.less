@import '~antd/dist/antd.less';
@import '~styles/variables.less';

body {
  margin: 0;
  font-family: 'Roboto', 'Segoe UI', sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: @white;
  background-color: @background;
  padding: @padding-big;
  overflow: hidden;
}

::-webkit-scrollbar {
  width: 8px;
  height: 8px;
}
::-webkit-scrollbar-button {
  width: 0px;
  height: 0px;
}
::-webkit-scrollbar-thumb {
  background: #e1e1e1;
  border: 0px none #ffffff;
  border-radius: 50px;
}
::-webkit-scrollbar-thumb:hover {
  background: #ffffff;
}
::-webkit-scrollbar-thumb:active {
  background: #000000;
}
::-webkit-scrollbar-track {
  background: #666666;
  border: 0px none #ffffff;
  border-radius: 50px;
}
::-webkit-scrollbar-track:hover {
  background: #666666;
}
::-webkit-scrollbar-track:active {
  background: #333333;
}
::-webkit-scrollbar-corner {
  background: transparent;
}

#root, .app {
  height: 100%;
}

.header-text > sup {
  color: @primary-color;
  text-decoration: underline;
}

.app {
  display: flex;
  &-content {
    width: calc(100% - @sidebar-width);

    main {
      height: 100%;
    }

    &-user-space {
      display: flex;
      width: 100%;
      justify-content: flex-end;
    }

    &-routes {
      height: 100%;
      overflow-y: auto;

      &-container {
        padding-bottom: 80px;
      }
    }
  }
}