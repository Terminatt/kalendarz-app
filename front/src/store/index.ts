import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { modalSlice } from './modals/slice';
import { userSlice } from './user/slice';

const rootReducer = combineReducers({
    user: userSlice.reducer,
    modal: modalSlice.reducer,
});

export const store = configureStore({
    reducer: rootReducer,
});

export type RootState = ReturnType<typeof rootReducer>
export type AppDispatch = typeof store.dispatch
