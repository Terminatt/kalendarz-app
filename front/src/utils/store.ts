import { RejectResponse } from '@generics/generics';
import { createAsyncThunk, Dispatch } from '@reduxjs/toolkit';
import { AxiosResponse, AxiosError } from 'axios';

function isAxiosError(err: AxiosError | Error | unknown): err is AxiosError {
    return !!(err as AxiosError).response;
}

export interface CustomAsyncThunkResponse<Data = void> {
  data: Data,
  successMessage?: string
}

abstract class StoreUtils {
    static createCustomAsyncThunk<Payload, Response>(
        prefix: string,
        options: {
          request: (payload: Payload) => Promise<AxiosResponse<Response>>,
          successMessage: string,
          errorMessage: string
      },
    ) {
        return createAsyncThunk<CustomAsyncThunkResponse<Response>, Payload, { rejectValue: RejectResponse }>(
            prefix,
            async (payload, { rejectWithValue }) => {
                const { request, successMessage, errorMessage } = options;

                try {
                    const response = await request(payload);
                    const { data } = response;

                    return { data, successMessage };
                } catch (error) {
                    if (isAxiosError(error)) {
                        return rejectWithValue({ error, errorMessage });
                    }
                    return rejectWithValue({ errorMessage });
                }
            },
        );
    }
}

export default StoreUtils;
