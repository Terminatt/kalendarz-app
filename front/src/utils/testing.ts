export function matchMedia(): void {
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation((query: string) => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(),
            removeListener: jest.fn(),
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });
}
