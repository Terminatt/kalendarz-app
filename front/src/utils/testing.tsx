import React, { ComponentType } from 'react';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import { RootState, rootReducer } from '@store/index';

export function matchMedia(): void {
    Object.defineProperty(window, 'matchMedia', {
        writable: true,
        value: jest.fn().mockImplementation((query: string) => ({
            matches: false,
            media: query,
            onchange: null,
            addListener: jest.fn(),
            removeListener: jest.fn(),
            addEventListener: jest.fn(),
            removeEventListener: jest.fn(),
            dispatchEvent: jest.fn(),
        })),
    });
}
export function isTestingEnv(): boolean {
    return process.env.JEST_WORKER_ID !== undefined || process.env.NODE_ENV === 'test';
}

export interface CustomOptions extends RenderOptions {
    preloadedState: RootState;
}

function render(
    ui: React.ReactElement,
    options?: CustomOptions,
) {
    const store = configureStore({ reducer: rootReducer, preloadedState: options?.preloadedState });
    function Wrapper({ children }: {children: React.ReactNode}) {
        return <Provider store={store}>{children}</Provider>;
    }

    return { element: rtlRender(ui, { wrapper: Wrapper as ComponentType, ...options }), store };
}

// re-export everything
export * from '@testing-library/react';
// override render method
export { render };
