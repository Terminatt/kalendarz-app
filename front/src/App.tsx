import ReservationSearch from '@components/ReservationSearch/ReservationSearch';
import Sidebar from '@components/Sidebar/Sidebar';
import React from 'react';

import './App.less';

const App: React.FC = () => (
    <div className="app">
        <Sidebar bottom={<ReservationSearch />} />
    </div>
);

export default App;
