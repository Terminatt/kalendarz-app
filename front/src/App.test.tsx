import React from 'react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router';
import MockDate from 'mockdate';
import App from './App';
import { store } from './store';
import renderer from 'react-test-renderer';
import { matchMedia } from '@utils/testing';

describe('App Component', () => {
    beforeEach(() => {
        matchMedia();
        MockDate.set('2022-02-20');
    });

    afterEach(() => {
        MockDate.reset();
    })
    
    it('matches the snapshot', () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <MemoryRouter>
                        <App />
                    </MemoryRouter>
                </Provider>,
            )
            .toJSON();
        
        expect(tree).toMatchSnapshot();
    });
});
