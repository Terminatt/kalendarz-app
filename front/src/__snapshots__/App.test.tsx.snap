// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches the snapshot 1`] = `
<div
  className="app"
>
  <div
    className="app-sidebar"
  >
    <div />
  </div>
  <div
    className="app-content"
  >
    <main>
      <div
        className="app-content-bar"
      >
        <div
          className="signing"
        >
          <button
            className="ant-btn custom-btn custom-btn-clear signing-login"
            onClick={[Function]}
            type="button"
          >
            <span>
              Zaloguj się
            </span>
          </button>
          <div
            className="signing-register"
          >
            <button
              className="ant-btn custom-btn custom-btn-primary"
              onClick={[Function]}
              type="button"
            >
              <span>
                Rejestracja
              </span>
            </button>
            <div
              className="signing-forget-password"
            >
              Nie pamiętasz hasła?
            </div>
          </div>
        </div>
        <button
          aria-label="otwórz nawigacje"
          className="ant-btn custom-btn custom-btn-icon app-content-bar-hamburger"
          onClick={[Function]}
          type="button"
        >
          <span
            aria-label="menu"
            className="anticon anticon-menu"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="menu"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
              />
            </svg>
          </span>
        </button>
      </div>
      <div
        className="app-content-routes"
      >
        <div
          className="home"
        >
          <div
            className="home-content"
          >
            <div
              className="calendar home-content-calendar"
            >
              <div
                className="ant-divider ant-divider-horizontal ant-divider-with-text ant-divider-with-text-center huge-divider calendar-divider"
                role="separator"
              >
                <span
                  className="ant-divider-inner-text"
                >
                  2022
                </span>
              </div>
              <div
                className="calendar-switcher"
              >
                <div
                  className="switcher"
                >
                  <button
                    className="ant-btn transparent-btn switcher-btn switcher-btn-left"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-label="left"
                      className="anticon anticon-left"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="left"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
                        />
                      </svg>
                    </span>
                  </button>
                  <span
                    className="switcher-selected"
                  >
                    Luty
                  </span>
                  <button
                    className="ant-btn transparent-btn switcher-btn switcher-btn-right"
                    onClick={[Function]}
                    type="button"
                  >
                    <span
                      aria-label="right"
                      className="anticon anticon-right"
                      role="img"
                    >
                      <svg
                        aria-hidden="true"
                        data-icon="right"
                        fill="currentColor"
                        focusable="false"
                        height="1em"
                        viewBox="64 64 896 896"
                        width="1em"
                      >
                        <path
                          d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                        />
                      </svg>
                    </span>
                  </button>
                </div>
              </div>
              <div
                className="calendar-content"
              >
                <div
                  className="calendar-content-header"
                >
                  <h2>
                    Wybierz dzień
                  </h2>
                </div>
                <div
                  className="calendar-content-dates"
                >
                  <button
                    className="calendar-item calendar-item-another-month-day"
                    disabled={true}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      31
                    </div>
                    <div>
                      Pn
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      01
                    </div>
                    <div>
                      Wt
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      02
                    </div>
                    <div>
                      Śr
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      03
                    </div>
                    <div>
                      Czw
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      04
                    </div>
                    <div>
                      Pt
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      05
                    </div>
                    <div>
                      Sb
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      06
                    </div>
                    <div>
                      Nd
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      07
                    </div>
                    <div>
                      Pn
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      08
                    </div>
                    <div>
                      Wt
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      09
                    </div>
                    <div>
                      Śr
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      10
                    </div>
                    <div>
                      Czw
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      11
                    </div>
                    <div>
                      Pt
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      12
                    </div>
                    <div>
                      Sb
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      13
                    </div>
                    <div>
                      Nd
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      14
                    </div>
                    <div>
                      Pn
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      15
                    </div>
                    <div>
                      Wt
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      16
                    </div>
                    <div>
                      Śr
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      17
                    </div>
                    <div>
                      Czw
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      18
                    </div>
                    <div>
                      Pt
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-before-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      19
                    </div>
                    <div>
                      Sb
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-today"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      20
                    </div>
                    <div>
                      Nd
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-normal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      21
                    </div>
                    <div>
                      Pn
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-normal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      22
                    </div>
                    <div>
                      Wt
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-normal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      23
                    </div>
                    <div>
                      Śr
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-normal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      24
                    </div>
                    <div>
                      Czw
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-normal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      25
                    </div>
                    <div>
                      Pt
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-dayoff"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      26
                    </div>
                    <div>
                      Sb
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-dayoff"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      27
                    </div>
                    <div>
                      Nd
                    </div>
                  </button>
                  <button
                    className="calendar-item calendar-item-normal"
                    disabled={false}
                    onClick={[Function]}
                    type="button"
                  >
                    <div>
                      28
                    </div>
                    <div>
                      Pn
                    </div>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>
</div>
`;
